scipy/cluster/vq.py

```diff
@@ -208,17 +212,21 @@ def vq(obs, code_book, check_finite=True):
     code_book = _asarray(code_book, xp=xp, check_finite=check_finite)
     ct = xp.result_type(obs, code_book)
 
-    c_obs = xp.astype(obs, ct, copy=False)
-    c_code_book = xp.astype(code_book, ct, copy=False)
-
     if xp.isdtype(ct, kind='real floating'):
+        c_obs = xp.astype(obs, ct, copy=False)
+        c_code_book = xp.astype(code_book, ct, copy=False)
         c_obs = np.asarray(c_obs)
         c_code_book = np.asarray(c_code_book)
         result = _vq.vq(c_obs, c_code_book)
         return xp.asarray(result[0]), xp.asarray(result[1])
     return py_vq(obs, code_book, check_finite=False)
 
 
+# Workaround to https://github.com/numpy/numpydoc/issues/624
+whiten.__doc__ = whiten.__doc__.replace("check_finite : ", "\ncheck_finite : ")
+vq.__doc__ = vq.__doc__.replace("check_finite : ", "\ncheck_finite : ")
```

Comment:
- https://github.com/numpy/numpydoc/issues/624

Without this, `spin smoke-docs` crashes. Could not think of anything less ugly.

Reply:
Adding doctests into parameter description is a little exotic. Not very surprising that it runs into a hiccup or several TBH.

Reply:
Removed doctest in favour of a simple `::` block.